runtime/server/repos.go

```diff
@@ -57,16 +58,34 @@ func (s *Server) WatchFiles(req *runtimev1.WatchFilesRequest, ss runtimev1.Runti
 		return err
 	}
 
-	return repo.Watch(ss.Context(), req.Replay, func(event drivers.WatchEvent) error {
-		if !event.Dir {
-			err := ss.Send(&runtimev1.WatchFilesResponse{
-				Event: event.Type,
-				Path:  event.Path,
+	if req.Replay {
+		paths, err := repo.ListRecursive(ss.Context(), req.InstanceId, "**")
+		if err != nil {
+			return err
+		}
+		for _, p := range paths {
+			err = ss.Send(&runtimev1.WatchFilesResponse{
+				Event: runtimev1.FileEvent_FILE_EVENT_WRITE,
+				Path:  p,
 			})
+			if err != nil {
+				return err
+			}
+		}
+	}
 
-			return err
+	return repo.Watch(ss.Context(), "", func(events []drivers.WatchEvent) {
+		for _, event := range events {
+			if !event.Dir {
+				err := ss.Send(&runtimev1.WatchFilesResponse{
```

Comment:
should we add batches to the response too?

Reply:
It might make sense, depends what the frontend team would prefer. Will ask them and then we can make the change in a follow up PR if needed